mesonbuild/wrap/wrap.py

```diff
@@ -66,16 +67,23 @@ def whitelist_wrapdb(urlstr: str) -> urllib.parse.ParseResult:
         raise WrapException(f'WrapDB did not have expected SSL https url, instead got {urlstr}')
     return url
 
-def open_wrapdburl(urlstring: str, allow_insecure: bool = False, have_opt: bool = False) -> 'http.client.HTTPResponse':
+def open_wrapdburl(urlstring: str, allow_insecure: bool = False, have_opt: bool = False, allow_compression: bool = False) -> 'http.client.HTTPResponse':
     if have_opt:
         insecure_msg = '\n\n    To allow connecting anyway, pass `--allow-insecure`.'
     else:
         insecure_msg = ''
 
+    def do_urlopen(url: urllib.parse.ParseResult) -> 'http.client.HTTPResponse':
```

Comment:
```suggestion
    def do_urlopen(url: urllib.parse.ParseResult) -> http.client.HTTPResponse:
```

iirc if possible, we should use non-quoted type names for new code.

Reply:
Fixed the method signatures.  `cast()` still needs a string because `http.client` isn't imported at runtime.

Reply:
Note that the reason to prefer non-quoted type names is because inside of a type annotation, it is redundant due to `from __future__ import annotations` which has the semantic effect of implicitly quoting all annotations for you.

So the general rule is still for cast() to always quote, because we want typing to be as close to zero runtime cost as possible, and quoting the argument to cast() means python doesn't need to do name lookup / object creation (and for non-dotted names it can do string interning).